packages/itwin/grouping-mapping-widget/src/widget/components/GroupAction.tsx

```diff
@@ -238,95 +211,116 @@ export const GroupAction = (props: GroupActionProps) => {
-                details.groupName,
-                NAME_REQUIREMENTS,
-              )
-                ? "negative"
-                : undefined
-            }
-            onBlur={() => {
-              validator.showMessageFor("groupName");
-            }}
-            onBlurCapture={(event) => {
-              handleInputChange(event, details, setDetails);
-              validator.showMessageFor("groupName");
-            }}
-          />
-          <LabeledInput
-            id='description'
-            name='description'
-            label='Description'
-            value={details.description}
-            onChange={(event) => {
-              handleInputChange(event, details, setDetails);
-            }}
+      <div className="gmw-group-add-modify-container">
+        <Fieldset legend={displayStrings.groupBy} className={
+          currentStep === GroupActionStep.QueryBuilder
+            ? "gmw-query-builder-container"
+            : "gmw-hide"
+        }>
+          <span>
+            <Label htmlFor='query-combo-input'>Query Generation Tool</Label>
+            <ComboBox
+              value={queryGenerationType}
+              inputProps={{
+                id: "query-combo-input",
+              }}
+              options={getOptions}
+              onChange={onChange}
+            />
+          </span>
+          <QueryBuilderCustomUI
+            queryGenerationType={queryGenerationType}
+            groupUIs={groupUIs}
+            isUpdating={isUpdating}
+            resetView={resetView}
+            setQuery={setQuery}
           />
         </Fieldset>
-        <Fieldset legend={displayStrings.groupBy} className='gmw-query-builder-container'>
-          <RadioTileGroup className='gmw-radio-group-tile' required>
-            {groupUIs.map((ext) => getRadioTileComponent(ext.icon ?? <SvgAdd />, ext.name, ext.displayLabel))}
-          </RadioTileGroup>
-          {queryGenerationType && createQueryBuilderComponent()}
+        <Fieldset
+          legend={displayStrings.groupDetails}
+          className={
+            currentStep === GroupActionStep.GroupDetails
+              ? "gmw-group-details"
+              : "gmw-hide"
+          }>
+          <GroupDetails
+            details={details}
+            setDetails={setDetails}
+            validator={validator}
+          />
         </Fieldset>
       </div>
-      <ActionPanel
-        onSave={async () => {
-          await save();
-        }}
-        onCancel={props.onClickCancel ? async () => {
-          Presentation.selection.clearSelection(
-            "GroupingMappingWidget",
-            iModelConnection,
-          );
-          props.onClickCancel && props.onClickCancel();
-        } : undefined}
-        isSavingDisabled={isBlockingActions}
-        isLoading={isLoading}
-      />
+      <div className='gmw-action-panel'>
+        {isLoading &&
+          <LoadingSpinner />
+        }
+        {currentStep !== GroupActionStep.QueryBuilder ?
```

Comment:
It would be best to have separate components for each step instead of relying on a bunch of ternary operators to display the right buttons / titles / classes, etc.

Reply:
Separated components.